flake.nix

```diff
@@ -33,6 +37,9 @@
       (system:
       let
         pkgs = import inputs.nixpkgs { inherit system; };
+        tools = {
+          hlint = hydraProject.pkgs.haskell-nix.tool hydraProject.compiler "hlint" "3.8";
+        };
```

Comment:
I would prefer it the other way around. i.e. put `tools` into the `hydraProject` expression (if this is possible?)

Reply:
I would first bring the pkgs expression up to the flake.nix before attempting that. I don't *think* there is a need to match on more than one pkgs.

Reply:
So you mean the haskell.nix enhanced packages (here `hydraProject.pkgs`) merged into the nixpkgs `pkgs`?

That would make sense to me as well. It just seemed odd that the `hydraProject.compiler` is passed into this expression again, while it would be already available in there.

Reply:
Yes, resolve the pkgs expression first with the iohk overlays included as the first binding and then use that in the whole program.